.github/workflows/PR.yml

```diff
@@ -0,0 +1,201 @@
+name: PR Build and Test
```

Comment:
Instead of duplicating the workflow file, can't we just use something like https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsif ?

Reply:
Yes that is possible. It would complicate the workflow only a little bit.
My idea behind the 2 different workflows is the clarity we get from that. 
You see directly, that the workflows are different for a push and a PR.

Reply:
And to be closer to the workflows indy-node and plenum use.

Reply:
IMO: The workflow files are already quite large and maintaining duplicated code isn't fun, so I vote for one (slightly more complicated) file. :man_shrugging: 

@Echsecutor and @c2bo any opinions on that?